sql/index/pilosa/driver.go

```diff
@@ -140,12 +142,50 @@ var (
 	errDeletePilosaIndex = errors.NewKind("error deleting pilosa index %s: %s")
 )
 
+type bitBatch struct {
+	size uint64
+	bits []pilosa.Bit
+	pos  uint64
+}
+
+func newBitBatch(size uint64) *bitBatch {
+	b := &bitBatch{size: size}
+	b.Clean()
+
+	return b
+}
+
+func (b *bitBatch) Clean() {
+	b.bits = make([]pilosa.Bit, 0, b.size)
```

Comment:
I think here we can just `b.bits = nil` - this will release the underlying array to the GC.

Reply:
This will also release the previous array to GC as it is not reachable anymore. I've also preferred to allocate the minimum capacity it will have (batch size) to keep the number of slice grow low.